ext/rotoscope/rotoscope.c

```diff
@@ -215,17 +217,18 @@ static void event_hook(VALUE tpval, void *data)
     return;
 
   rs_tracepoint_t trace = extract_full_tracevals(trace_arg, &trace_path);
-  gzprintf(config->log, RS_CSV_FORMAT, RS_CSV_VALUES(trace));
+  fprintf(config->log, RS_CSV_FORMAT, RS_CSV_VALUES(trace));
 }
 
-static void close_gz_handle(Rotoscope *config)
+static void close_log_handle(Rotoscope *config)
 {
   if (config->log)
   {
     if (!in_fork(config))
     {
-      gzclose(config->log);
+      fclose(config->log);
     }
+    close(fileno(config->log));
```

Comment:
This stems from a strange edge case actually. In the case of `fork do ... end` in ruby, we end up with a copy of the same FD in two different processes. If we are in the original copy, we want to invoke `fclose` to flush the buffer and close out/free the descriptor, which is what the earlier branch does.

…BUT if we are in the forked process, we don't want to flush the buffer in case the fork occurred while data had yet to be flushed from the buffer to the file itself. `fpurge` does this in a more obvious way, but unfortunately that's a non-standard feature only available in BSD. I found this out the hard way from the `fork` test we have in the repo that was duplicating data that had been sitting in the buffer, pre-fork.

`close` will close/free the file descriptor, but without invoking any `at_exit` handlers. `fopen` registers a hook to flush the buffer on close, so by using `close` we avoid the accidental flush 🙃